sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -100,39 +100,75 @@ private class CosmosRowConverter(
 
     def fromRowToObjectNode(row: Row): ObjectNode = {
 
-        if (row.schema.contains(StructField(CosmosTableSchemaInferrer.RawJsonBodyAttributeName, StringType))){
-            // Special case when the reader read the rawJson
-            val rawJson = row.getAs[String](CosmosTableSchemaInferrer.RawJsonBodyAttributeName)
-            objectMapper.readTree(rawJson).asInstanceOf[ObjectNode]
-        }
-        else {
-            val objectNode: ObjectNode = objectMapper.createObjectNode()
-            row.schema.fields.zipWithIndex.foreach({
-              case (field, i) =>
-                field.dataType match {
-                  case _: NullType => putNullConditionally(objectNode, field.name)
-                  case _ if row.isNullAt(i) => putNullConditionally(objectNode, field.name)
-                  case _ =>
-                    val nodeOpt = convertSparkDataTypeToJsonNode(field.dataType, row.get(i))
-                    if (nodeOpt.isDefined) {
-                      objectNode.set(field.name, nodeOpt.get)
-                    }
+      val rawBodyFieldName = if (row.schema.names.contains(CosmosTableSchemaInferrer.RawJsonBodyAttributeName) &&
+        row.schema.apply(CosmosTableSchemaInferrer.RawJsonBodyAttributeName).dataType.isInstanceOf[StringType]) {
```

Comment:
why checking for StringType here and below ?

Reply:
Different column names - both need to be of type StringType so that we can use them as raw-json columns

Reply:
+1 can't we just do the `contains` similar to what was before?